core/editable.js

```diff
@@ -1042,12 +1042,16 @@
 								( selected = getSelectedTableList( sel ) ) ) {
 							// Make undo snapshot.
 							editor.fire( 'saveSnapshot' );
+							if ( !selected.isMultipleSelectedList ) {
+								// Delete any element that 'hasLayout' (e.g. hr,table) in IE8 will
+								// break up the selection, safely manage it here. (https://dev.ckeditor.com/ticket/4795)
+								range.moveToPosition( selected.selection, CKEDITOR.POSITION_BEFORE_START );
+								// Remove the control manually.
+								selected.selection.remove();
+							} else {
+								selected.selection.deleteContents();
+							}
```

Comment:
Negation statements always require a little bit more focus than plain value. So we could change this code as below:
```suggestion
							if ( selected.isMultipleSelectedList ) {
							    selected.selection.deleteContents();
							} else {
								// Delete any element that 'hasLayout' (e.g. hr,table) in IE8 will
								// break up the selection, safely manage it here. (https://dev.ckeditor.com/ticket/4795)
								range.moveToPosition( selected.selection, CKEDITOR.POSITION_BEFORE_START );
								// Remove the control manually.
								selected.selection.remove();
							}
```